Src/Base/AMReX_DistributionMapping.cpp

```diff
@@ -1641,6 +1818,145 @@ DistributionMapping::makeSFC (const Vector<Real>& rcost, const BoxArray& ba, Rea
     return r;
 }
 
+DistributionMapping
+DistributionMapping::makeSFC (const LayoutData<Real>& rcost_local,
```

Comment:
The first part of this function is very similar to the first part of the first new function.  Maybe this part can be put into a new function.

Reply:
I've added a function ParallelDescriptor::GatherLayoutData which collections LayoutData local values to a vector on root process, this eliminate the code repetition.